modules/tinymce/src/core/main/ts/delete/DeleteUtils.ts

```diff
@@ -78,8 +71,11 @@ const deleteRangeContents = (editor: Editor, rng: Range, root: SugarElement<HTML
   rng.deleteContents();
   // Pad the last block node
   const lastNode = freefallRtl(root).getOr(root);
-  const lastBlock = SugarElement.fromDom(editor.dom.getParent(lastNode.dom, editor.dom.isBlock));
-  if (Empty.isEmpty(lastBlock)) {
+  const lastBlock = SugarElement.fromDom(editor.dom.getParent(lastNode.dom, editor.dom.isBlock) ?? root.dom);
+  // If the block is the editor body then we need to insert the root block as well
+  if (lastBlock.dom === editor.getBody()) {
+    paddEmptyBody(editor, moveSelection);
```

Comment:
@ltrouton pointed out today that this function already existed, so the `ForceBlocks.insertNewLine` code is no longer required for this fix. However, as it does remove duplication I think it's worth keeping. If anyone would prefer it to be reverted though let me know and I'll do it.